src/aliceVision/sfm/View.hpp

```diff
@@ -251,12 +251,12 @@ class View
    * @brief Get the corresponding "FocalLength" metadata value
    * @return the metadata value float or -1 if no corresponding value
    */
-  float getMetadataFocalLength() const
+  double getMetadataFocalLength() const
   {
     if(hasMetadata("Exif:FocalLength"))
-      return std::stof(getMetadata("Exif:FocalLength"));
+      return std::stod(getMetadata("Exif:FocalLength"));
```

Comment:
Can you add a check to return the value only if it's not 0?
If the value is zero we consider that the metadata doesn't exist.

Same for FocalLengthIn35mm.